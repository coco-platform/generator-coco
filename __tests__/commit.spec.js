/**
 * @description - coco:prettier generator
 * @author - huang.jian <hjj491229492@hotmail.com>
 */

// packages
const path = require('path');
const fs = require('fs');
const renderer = require('yeoman-test');
const assert = require('yeoman-assert');

describe('coco:commit', () => {
  it('should setup commitizen environment', () => {
    const commit = path.resolve(__dirname, '../generators/commit');

    return renderer
      .run(commit)
      .inTmpDir((dir) => {
        const template = path.resolve(__dirname, '__fixtures__/package.json');
        const destiny = path.resolve(dir, 'package.json');

        fs.copyFileSync(template, destiny);
      })
      .then(() => {
        assert.file(['.commitlintrc.yml']);

        const meta = fs.readFileSync('package.json', 'utf8');

        expect(meta).toMatchSnapshot();
      });
  });
});
