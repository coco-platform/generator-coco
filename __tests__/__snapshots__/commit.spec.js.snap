// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coco:commit should setup commitizen environment 1`] = `
"{
  \\"name\\": \\"generator-coco\\",
  \\"version\\": \\"0.2.0\\",
  \\"description\\": \\"NPM browser compatible package.\\",
  \\"keywords\\": [
    \\"typescript\\",
    \\"package\\"
  ],
  \\"author\\": \\"huang.jian <hjj491229492@hotmail.com>\\",
  \\"contributors\\": [
    {
      \\"name\\": \\"huang.jian\\",
      \\"email\\": \\"hjj491229492@hotmail.com\\"
    },
    {
      \\"name\\": \\"yang.yuncai\\",
      \\"email\\": \\"383235388@qq.com\\"
    }
  ],
  \\"license\\": \\"MIT\\",
  \\"dependencies\\": {},
  \\"devDependencies\\": {},
  \\"config\\": {
    \\"commitizen\\": {
      \\"path\\": \\"cz-conventional-changelog\\"
    }
  },
  \\"husky\\": {
    \\"hooks\\": {
      \\"prepare-commit-msg\\": \\"exec < /dev/tty && git cz --hook\\",
      \\"commit-msg\\": \\"commit -E HUSKY_GIT_PARAMS\\"
    }
  }
}
"
`;
